lib/logging.dart

```diff
@@ -364,3 +366,187 @@ class LogRecord {
   @override
   String toString() => '[${level.name}] $loggerName: $message';
 }
+
+typedef _ServiceExtensionCallback = Future<Map<String, dynamic>> Function(
+    Map<String, String> parameters);
+
+/// A shared manager instance whose recorded messages are logged to the
+/// developer console.
+final LogManager logManager = LogManager()..onRecord.listen((record) {
+  developer.log(
+    record.message,
+    name: record.loggerName,
+    time: record.time,
+    error: record.error,
+    level: record.level.value,
+    stackTrace: record.stackTrace,
+  );
+});
+
+/// Log Managers record messages only written to loggers explicitly enabled by
+/// name.
+///
+/// For example, in the given source, only messages to `loggerA` will be sent to
+/// the `onRecord` stream.
+///
+/// ```
+/// var loggerA = Logger('loggerA');
+/// var loggerA = Logger('loggerB');
+///
+/// var manager = LogManager()..enableLogging('loggerA');
+///
+/// loggerA.info('hello from logger A'); // recorded
+/// loggerB.info('here goes nothing');   // ignored
+/// ```
+///
+/// Log managers can be created but the provided [logManager] should suffice
+/// for most purposes.  The shared manager redirects recorded messages to the
+/// developer log.
+///
+/// **Important note:** managed loggers respect logger `root` level configuration.
+/// For example, in the following, `logger` only reports `severe` messages
+/// as expected.
+///
+/// ```
+/// Logger.root.level = Level.SEVERE;
+///
+/// ...
+///
+/// var logger = Logger('my.logger');
+/// logManager.enable('my.logger');
+/// try {
+///   logger.info('doing something'); // ignored
+///   ...
+/// } catch(e, stackTrace) {
+///   logger.severe('uh-oh!, e, stackTrace); // recorded
+/// }
+/// ```
+///
+/// * To create loggers, see [Logger].
+/// * To enable or disable a logger, use [enableLogging].
+/// * To query channel enablement, use [shouldLog].
+class LogManager {
+  static bool _initialized;
+
+  final Set<String> _enabledLoggers = Set<String>();
+
+  final StreamController<LogRecord> _logController =
+      StreamController.broadcast(sync: true);
+  final StreamController<String> _loggerAddedBroadcaster =
+      StreamController.broadcast();
+  final StreamController<String> _loggerEnabledBroadcaster =
+      StreamController.broadcast();
+
+  LogManager() {
+    Logger.root.onRecord.listen((record) {
+      if (shouldLog(record.loggerName)) {
+        _logController.add(record);
+      }
+    });
+  }
+
+  /// Returns a stream of messages added to loggers enabled by this manager.
+  Stream<LogRecord> get onRecord => _logController.stream;
```

Comment:
Is this exposed for testing or for users? If you are implementing the full Logger api, might make sense to implement Logger.